File: scripts/check-erp-data.mjs
import Database from "better-sqlite3";
import { readFileSync, existsSync } from "node:fs";

const paths = process.argv.slice(2);
if (!paths.length) {
  console.error("Usage: node check-erp-data.mjs <sqlite-path>...");
  process.exit(1);
}

for (const dbPath of paths) {
  if (!existsSync(dbPath)) {
    console.log(`=== ${dbPath} === MISSING`);
    continue;
  }
  const db = new Database(dbPath, { readonly: true });
  const row = db.prepare("SELECT version, updated_at, payload FROM erp_state WHERE id = 1").get();
  if (!row) {
    console.log(`=== ${dbPath} === no erp_state`);
    db.close();
    continue;
  }
  const payload = JSON.parse(row.payload);
  console.log(`=== ${dbPath} ===`);
  console.log(`version=${row.version} updated=${row.updated_at} payload_bytes=${row.payload.length}`);
  for (const key of [
    "sales",
    "workers",
    "workerMonthlyActualVouchers",
    "workerPaymentRecords",
    "bankTransactions",
    "workerPayoutVouchers",
  ]) {
    const value = payload[key];
    console.log(`  ${key}: ${Array.isArray(value) ? value.length : String(value)}`);
  }
  const memos = (payload.workers || []).filter((w) => String(w.monthlyPaymentMemo || "").trim()).length;
  console.log(`  workers with monthlyPaymentMemo: ${memos}`);
  db.close();
}

File: scripts/check-erp-data.py
#!/usr/bin/env python3
import json
import os
import sqlite3
import sys

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
os.chdir(ROOT)

paths = sys.argv[1:] or ["data/erp.sqlite", "data/erp.sqlite.bak-202606010611"]
keys = [
    "sales",
    "workers",
    "workerMonthlyActualVouchers",
    "workerPaymentRecords",
    "bankTransactions",
]

for db_path in paths:
    print(f"=== {db_path} ===")
    if not os.path.exists(db_path):
        print("  MISSING")
        continue
    con = sqlite3.connect(db_path)
    row = con.execute("SELECT version, updated_at, payload FROM erp_state WHERE id = 1").fetchone()
    if not row:
        print("  no erp_state row")
        con.close()
        continue
    payload = json.loads(row[2])
    print(f"  version={row[0]} updated={row[1]}")
    for key in keys:
        value = payload.get(key) or []
        print(f"  {key}: {len(value)}")
    memos = sum(
        1
        for worker in (payload.get("workers") or [])
        if str(worker.get("monthlyPaymentMemo") or "").strip()
    )
    print(f"  workers with monthlyPaymentMemo: {memos}")
    con.close()
